gemd/entity/bounds_validation.py

```diff
@@ -0,0 +1,43 @@
+from enum import IntEnum
+from contextlib import contextmanager
+
+
+class WarningLevel(IntEnum):
+    """
+    Control the behavior for warnings/errors around template validations.
+
+    IGNORE: Do not check if values are consistent with bounds.
+    WARNING: Accept bad values and issue a warning saying as much.
+    FATAL: Raise an exception when trying to set an inconsistent value.
+
+    """
+
+    IGNORE = 0
+    WARNING = 1
+    FATAL = 2
+
+
+BOUNDS_VALIDATION = WarningLevel.WARNING
+
+
+def get_validation_level() -> WarningLevel:
+    """Return the value of the BOUNDS_VALIDATION."""
+    return BOUNDS_VALIDATION
+
+
+def set_validation_level(level: WarningLevel):
+    """Set the value of the BOUNDS_VALIDATION."""
+    global BOUNDS_VALIDATION
+    BOUNDS_VALIDATION = WarningLevel(level)
+
+
+@contextmanager
+def validation_context(level: WarningLevel):
```

Comment:
Defining custom contexts with this decorator is something I haven't used before, so I'm mainly just curious.

Is this used in the form of `with validation_context(level): do things...` ?

If so, would naming it `validation_level` fit better gramatically?

Reply:
You are correct in how it is used, and your suggestion makes sense grammatically.  My only hesitation would be if having `get_validation_level`, `set_validation_level` and `validation_level` methods would be confusing.  I'm inclined to say `It'll be fine.`